### PR TITLE
Application facade tests

### PR DESCRIPTION
**Pull request by raonismaneoto**

### CODE REVIEW

**Comment by raonismaneoto** [comment]
There are some critical changes that must be done in the applicationFacade. I'll address them in other pr, to make it easier. So, there is no refact in the tested class in the pr.

File: src/test/java/cloud/fogbow/fns/core/ApplicationFacadeTest.java [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+        Mockito.verify(applicationFacade, Mockito.times(TestUtils.RUN_ONCE)).authorizeOrder(Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any());
+        Mockito.verify(federatedNetworkOrderController, Mockito.times(TestUtils.RUN_ONCE)).deleteFederatedNetwork(Mockito.any());
+    }
+
+    //test case: check if the method makes the expected calls.
+    @Test
+    public void testCreateCompute() throws FogbowException {
+        //setup
+        FederatedNetworkOrder order = testUtils.createFederatedNetwork(FEDERATED_NETWORK_ID, OrderState.FULFILLED);
+        Mockito.doReturn("127.0.0.1").when(order).getFreeIp();
```
Comment (ANKerD) [inline src/test/java/cloud/fogbow/fns/core/ApplicationFacadeTest.java]: Please move this magic String to a constant.

File: src/test/java/cloud/fogbow/fns/core/ApplicationFacadeTest.java [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+        Mockito.doReturn(order).when(federatedNetworkOrderController).getFederatedNetwork(Mockito.any());
+        Mockito.doNothing().when(applicationFacade).addUserDataToCompute(Mockito.any(), Mockito.any());
+        FederatedCompute compute = testUtils.createFederatedCompute(FEDERATED_NETWORK_ID);
+        PowerMockito.mockStatic(RedirectToRasUtil.class);
+        BDDMockito.given(RedirectToRasUtil.createAndSendRequestToRas(Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any(), Mockito.any()))
+            .willThrow(new RestClientException(TestUtils.EMPTY_STRING));
+
+        try {
+            //exercise
+            applicationFacade.createCompute(compute, TestUtils.FAKE_TOKEN);
```
Comment (ANKerD) [inline src/test/java/cloud/fogbow/fns/core/ApplicationFacadeTest.java]: I think this test should fail if no exception is thrown.

Comment (ANKerD) [reply src/test/java/cloud/fogbow/fns/core/ApplicationFacadeTest.java]: There are other methods in this class with the same issue, fix that, please.

File: src/main/java/cloud/fogbow/fns/core/ApplicationFacade.java [hunk 3]
```diff
@@ -156,7 +150,7 @@ public synchronized String createCompute(FederatedCompute federatedCompute, Stri
                     HttpMethod.POST, systemUserToken, String.class);
         } catch (RestClientException e) {
             responseEntity = ResponseEntity.status(HttpStatus.BAD_GATEWAY).
-                    body(Messages.Error.RESOURCE_ALLOCATION_SERVICE_DOES_NOT_RESPOND);
+                    body("{\"message\":\"" + Messages.Error.RESOURCE_ALLOCATION_SERVICE_DOES_NOT_RESPOND + "\"}");
```
Comment (wassoares) [inline src/main/java/cloud/fogbow/fns/core/ApplicationFacade.java]: Move this string to a constant of kind "{\"message\":%s}" and use String.format method.
Do the same for other occurrences.

File: src/test/java/cloud/fogbow/fns/TestUtils.java [hunk 4]
```diff
@@ -3,31 +3,39 @@
...
     @NotNull
     public FederatedNetworkOrder createFederatedNetwork(String id, OrderState state) {
         HashMap<String, MemberConfigurationState> allowedMembers = new HashMap<>();
         Queue<String> freedIps = new LinkedList<>();
         ArrayList<AssignedIp> computesIp = new ArrayList<>();
         computesIp.add(new AssignedIp(FAKE_COMPUTE_ID, FAKE_IP));
-        FederatedNetworkOrder federatedNetworkOrder = new FederatedNetworkOrder(id, user, MEMBER, MEMBER, CIDR,
-                "name", allowedMembers, freedIps, computesIp, state);
+        FederatedNetworkOrder federatedNetworkOrder = Mockito.spy(new FederatedNetworkOrder(id, user, MEMBER, MEMBER, CIDR,
+                "name", allowedMembers, freedIps, computesIp, state));
```
Comment (wassoares) [inline src/test/java/cloud/fogbow/fns/TestUtils.java]: Move the string "name" to a constant.

File: src/test/java/cloud/fogbow/fns/core/ApplicationFacadeTest.java [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+        } catch (FogbowException ex) {
+            //verify
+            Assert.assertEquals(Messages.Error.RESOURCE_ALLOCATION_SERVICE_DOES_NOT_RESPOND, ex.getMessage());
+        }
+        //verify
+        Mockito.verify(federatedNetworkOrderController, Mockito.times(TestUtils.RUN_ONCE)).getFederatedNetwork(Mockito.any());
+        Mockito.verify(applicationFacade, Mockito.times(TestUtils.RUN_ONCE)).addUserDataToCompute(Mockito.any(), Mockito.any());
+        PowerMockito.verifyStatic(RedirectToRasUtil.class, Mockito.times(TestUtils.RUN_ONCE));
+        RedirectToRasUtil.createAndSendRequestToRas("/" + Compute.COMPUTE_ENDPOINT, testUtils.gson.toJson(compute.getCompute()),
```
Comment (wassoares) [inline src/test/java/cloud/fogbow/fns/core/ApplicationFacadeTest.java]: Move this kind of string "/" or "" to a constant.
Do the same for other occurrences.

File: src/test/java/cloud/fogbow/fns/core/ApplicationFacadeTest.java [hunk 6]
```diff
@@ -0,0 +1,378 @@
...
+
+    //test case: check if the method throws an exception when the requester is not the order's owner
+    @Test(expected = UnauthorizedRequestException.class)//verify
+    public void testAuthorizeOrderWithDifferentRequester() throws FogbowException {
+        //setup
+        FederatedNetworkOrder order = testUtils.createFederatedNetwork(FEDERATED_NETWORK_ID, OrderState.FULFILLED);
+        order.setSystemUser(testUtils.user);
+        Mockito.doReturn(true).when(authPlugin).isAuthorized(Mockito.any(), Mockito.any());
+        //exercise
+        applicationFacade.authorizeOrder(new SystemUser("fake-id", "fake-name", "fake-provider"),
```
Comment (wassoares) [inline src/test/java/cloud/fogbow/fns/core/ApplicationFacadeTest.java]: Move the strings of this line to constants.

File: src/main/java/cloud/fogbow/fns/core/ApplicationFacade.java (hunk 3 above)
Comment (ANKerD) [reply src/main/java/cloud/fogbow/fns/core/ApplicationFacade.java]: Done.

File: src/test/java/cloud/fogbow/fns/TestUtils.java (hunk 4 above)
Comment (ANKerD) [reply src/test/java/cloud/fogbow/fns/TestUtils.java]: Removed.

File: src/test/java/cloud/fogbow/fns/core/ApplicationFacadeTest.java (hunk 5 above)
Comment (ANKerD) [reply src/test/java/cloud/fogbow/fns/core/ApplicationFacadeTest.java]: Fixed all occurrences of this pattern.

File: src/test/java/cloud/fogbow/fns/core/ApplicationFacadeTest.java (hunk 6 above)
Comment (ANKerD) [reply src/test/java/cloud/fogbow/fns/core/ApplicationFacadeTest.java]: Moved.

**Comment by raonismaneoto** [comment]
Deprecated

**Pull request closed without merging**